packages/smooth_app/lib/helpers/analytics_helper.dart

```diff
@@ -146,8 +147,16 @@ class AnalyticsHelper {
   }
 
   /// A UUID must be at least one 16 characters
-  static String? get uuid =>
-      kDebugMode ? 'smoothie-debug--' : OpenFoodAPIConfiguration.uuid;
+  static String? get uuid {
+    // if user opts out then track anonymously with userId containg zeros
+    if (!_allow) {
+      return '0' * 16;
+    } else if (kDebugMode) {
+      return 'smoothie_debug--';
+    } else {
+      return OpenFoodAPIConfiguration.uuid;
+    }
```

Comment:
```suggestion
    if (kDebugMode) {
      return 'smoothie_debug--';
    }
    // if user opts out then track anonymously with userId containg zeros
    if (!_allow) {
      return '0' * 16;
    }
    return OpenFoodAPIConfiguration.uuid;
```

I believe we want to track "real users", not developers debugging.
That's why I would put the `kDebugMode` check first.
In addition to that I don't like `else`s so as we `return` we can get rid of them, but it's just a matter of personal taste.
The most important here is to check `kDebugMode` first.

Reply:
If kDebugMode is checked first, the test always fails, because tests are always run in debug mode (always returns "smoothie_debug--". Do you have an idea how to overcome this? My first thought was setting an env-variable like TEST_MODE in the test and checking for it in the "get uuid"-method, but this seems dirty to me. But honestly I can't come up with something better.

Reply:
I understand your point, but you seem to be more interested in the test than in the code.
The code should match our needs, and that's why I would put the `kDebugMode` check first.

If there's no way to test it properly, let's not test it. Especially in that case, it's just a `if` we're checking. Of course a test would be better; it's not essential.

If really you insist, perhaps you could use `@visibleForTesting`. But I wouldn't put too much effort into it, and I would discard the test altogether.

Reply:
Thank you for your input. Am always interested in tests, because I come from a strict TDD-Household and the guys at the company I work in regard the tests as holy ;). But the most pragmatic way is to drop the test and it probably isn't worth it for like just 5 lines of code. I removed the test and the mockPackageInfo().

Reply:
Let's be pragmatic 😉